cmake/FindASIO.cmake

```diff
@@ -0,0 +1,97 @@
+#[=======================================================================[.rst:
+FindASIO
```

Comment:
The cmake linter preferred this case `FindASIO` but the cmake docs seem to follow `FindAsio` case. Any preference?

Reply:
I would go for cmake doc. It is what I saw most.